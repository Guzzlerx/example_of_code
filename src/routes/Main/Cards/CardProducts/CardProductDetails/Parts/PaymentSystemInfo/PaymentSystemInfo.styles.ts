import styled from 'styled-components';
import { COLOR_PALETTE } from 'constants/colors/colors';

export const PaymentSystemInfoContainer = styled.section`
  display: flex;
  justify-content: space-between;
  width: 100%;
  margin-bottom: 87px;

  & p:last-of-type {
    margin-top: 16px;
  }
`;

export const StyledInfoContainer = styled.div`
  display: flex;
  flex-direction: column;
  text-align: left;
`;

export const StyledTextContainer = styled.div`
  width: 416px;
  text-align: left;
`;

export const StyledSubtitle = styled.p`
  font-size: 14px;
  line-height: 20px;
  color: ${COLOR_PALETTE.gray700};
`;

export const StyledContainer = styled.div`
  display: flex;
  gap: 50px;
  margin-top: 32px;
`;

export const StyledIconWrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 16px;
`;

export const StyledPaymentLink = styled.a`
  font-size: 14px;
  font-weight: 700;
  line-height: 20px;
  color: ${COLOR_PALETTE.notification};
`;
