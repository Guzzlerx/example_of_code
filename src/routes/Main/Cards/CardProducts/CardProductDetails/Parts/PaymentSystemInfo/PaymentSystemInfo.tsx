import { useTranslation } from 'react-i18next';
import { AIcon } from 'components/AIcon/AIcon';
import { ATitle } from 'components/ATitle/ATitle';
import { ReactComponent as MastercardLogo } from 'assets/icons/mastercard_logo.svg';
import { ReactComponent as VISALogo } from 'assets/icons/visa_blue.svg';
import { ReactComponent as PaymentInfoLogo } from 'assets/image/cardDetailsInfo/man_with_many_cards.svg';
import { LinksEnum } from 'constants/links';
import {
  PaymentSystemInfoContainer,
  StyledContainer,
  StyledIconWrapper,
  StyledInfoContainer,
  StyledPaymentLink,
  StyledSubtitle,
  StyledTextContainer,
} from './PaymentSystemInfo.styles';

export const PaymentSystemInfo = () => {
  const { t } = useTranslation('cards');

  const translations = {
    title: t('card_details.payment_info.title'),
    firstSubtitle: t('card_details.payment_info.first_subtitle'),
    secondSubtitle: t('card_details.payment_info.second_subtitle'),
    visaText: t('card_details.payment_info.visa_text'),
    mastercardText: t('card_details.payment_info.mastercard_text'),
  };

  const CARDS_CONFIG = [
    {
      id: 0,
      title: 'VISA',
      link: LinksEnum.Visa,
      linkText: translations.visaText,
      logo: <VISALogo />,
    },
    {
      id: 1,
      title: 'Mastercard',
      link: LinksEnum.Mastercard,
      linkText: translations.mastercardText,
      logo: <MastercardLogo />,
    },
  ];

  return (
    <PaymentSystemInfoContainer>
      <StyledInfoContainer>
        <StyledTextContainer>
          <ATitle size={20} lineHeight={28} weight={700} marginBottom={24}>
            {translations.title}
          </ATitle>
          <StyledSubtitle>{translations.firstSubtitle}</StyledSubtitle>
          <StyledSubtitle>{translations.secondSubtitle}</StyledSubtitle>
        </StyledTextContainer>
        <StyledContainer>
          {CARDS_CONFIG.map(({ id, link, linkText, title, logo }) => (
            <StyledIconWrapper key={id}>
              <AIcon variant="paymentSystem" iconTitle={title}>
                {logo}
              </AIcon>
              <StyledPaymentLink href={link} target={'_blank'}>
                {linkText}
              </StyledPaymentLink>
            </StyledIconWrapper>
          ))}
        </StyledContainer>
      </StyledInfoContainer>
      <PaymentInfoLogo />
    </PaymentSystemInfoContainer>
  );
};
