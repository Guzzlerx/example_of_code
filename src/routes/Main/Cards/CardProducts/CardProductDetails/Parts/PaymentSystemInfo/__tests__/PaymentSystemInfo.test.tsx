import { RenderResult, render, screen } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { COLOR_PALETTE } from 'constants/colors/colors';
import i18n from 'utils/languages/i18nForTests';
import { PaymentSystemInfo } from '../PaymentSystemInfo';

describe('PaymentSystemInfo', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(
      <I18nextProvider i18n={i18n}>
        <PaymentSystemInfo />
      </I18nextProvider>,
    );
  });

  it('should render correctly', () => {
    expect(renderResult.container).toBeInTheDocument();
  });

  it('snapshot', () => {
    expect(renderResult.container).toMatchSnapshot();
  });

  it('should render component title correctly', () => {
    const element = screen.getByText(/Canâ€™t decide what payment system to choose/i);

    expect(element).toBeInTheDocument();
  });

  it('visa link has correct href and text', () => {
    const element = screen.getAllByRole('link')[0];

    expect(element).toHaveAttribute('href', 'https://www.visa.com/');
    expect(element).toHaveTextContent(/All VISA Infinite benefits/i);
  });

  it('mastercard link has correct href and text', () => {
    const element = screen.getAllByRole('link')[1];

    expect(element).toHaveAttribute('href', 'https://www.mastercard.com/');
    expect(element).toHaveTextContent(/All Mastercard World Elite benefits/i);
  });

  it('first subtitle renders', () => {
    const element = screen.getByText(/There is no need to decide/i);

    expect(element).toBeInTheDocument();
    expect(element).toHaveStyle(`color: ${COLOR_PALETTE.gray700}`);
  });

  it('second subtitle renders', () => {
    const element = screen.getByText(/1 account number, 2 cards with/i);

    expect(element).toBeInTheDocument();
    expect(element).toHaveStyle(`color: ${COLOR_PALETTE.gray700}`);
  });
});
