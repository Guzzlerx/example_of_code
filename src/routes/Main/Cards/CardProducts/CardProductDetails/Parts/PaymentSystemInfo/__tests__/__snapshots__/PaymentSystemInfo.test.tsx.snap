// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentSystemInfo snapshot 1`] = `
<div>
  <section
    class="sc-dkrFOg YiZDx"
  >
    <div
      class="sc-hLBbgP eZOVIq"
    >
      <div
        class="sc-eDvSVe bQVfVc"
      >
        <h2
          class="sc-gswNZR hQAgET"
          size="20"
        >
          Can’t decide what payment system to choose?
        </h2>
        <p
          class="sc-jSUZER fwytjE"
        >
          There is no need to decide. We’ll provide you maximum benefits from Mastercard World Elite and VISA Infinite.
        </p>
        <p
          class="sc-jSUZER fwytjE"
        >
          1 account number, 2 cards with different systems - one plastic and one virtual. Double your benefits and discounts.
        </p>
      </div>
      <div
        class="sc-gKPRtg eCkIZh"
      >
        <div
          class="sc-iBYQkv kQBwho"
        >
          <div
            class="sc-bcXHqe gZeIcz"
            title="VISA"
          >
            <svg>
              visa_blue.svg
            </svg>
          </div>
          <a
            class="sc-ftTHYK gkWsUx"
            href="https://www.visa.com/"
            target="_blank"
          >
            All VISA Infinite benefits
          </a>
        </div>
        <div
          class="sc-iBYQkv kQBwho"
        >
          <div
            class="sc-bcXHqe gZeIcz"
            title="Mastercard"
          >
            <svg>
              mastercard_logo.svg
            </svg>
          </div>
          <a
            class="sc-ftTHYK gkWsUx"
            href="https://www.mastercard.com/"
            target="_blank"
          >
            All Mastercard World Elite benefits
          </a>
        </div>
      </div>
    </div>
    <svg>
      man_with_many_cards.svg
    </svg>
  </section>
</div>
`;
